drivers/scheduler/k8s/k8s.go

```diff
@@ -944,6 +944,74 @@ func (k *k8s) Describe(ctx *scheduler.Context) (string, error) {
 	return buf.String(), nil
 }
 
+func (k *k8s) ScaleApplication(ctx *scheduler.Context, newScaleFactorMap map[string]int32) error {
+	logrus.Infof("Scale all Deployments")
+	k8sOps := k8s_ops.Instance()
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			dep, err := k8sOps.GetDeployment(obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+			newScaleFactor := newScaleFactorMap[obj.Name+"-dep"]
+			*dep.Spec.Replicas = newScaleFactor
+			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+				return fmt.Errorf("%v", &scheduler.ErrFailedToUpdateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to update Deployment: %v. Err: %v", obj.Name, err),
+				})
+			}
+			logrus.Infof("App %s scaled to %d successfully.", obj.Name, newScaleFactor)
+		}
+	}
+	return nil
+}
+
+func (k *k8s) ScaleStatefulSet(ctx *scheduler.Context, newScaleFactorMap map[string]int32) error {
+	logrus.Infof("Scale all Stateful sets")
+	k8sOps := k8s_ops.Instance()
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*apps_api.StatefulSet); ok {
+			dep, err := k8sOps.GetStatefulSet(obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+			newScaleFactor := newScaleFactorMap[obj.Name+"-ss"]
+			*dep.Spec.Replicas = newScaleFactor
+			if _, err := k8sOps.UpdateStatefulSet(dep); err != nil {
+				return fmt.Errorf("%v", &scheduler.ErrFailedToUpdateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to update Deployment: %v. Err: %v", obj.Name, err),
+				})
+			}
+			logrus.Infof("App %s scaled to %d successfully.", obj.Name, int(newScaleFactor))
+		}
+	}
+	return nil
+}
+
+func (k *k8s) GetNewScaleFactorMap(ctx *scheduler.Context, factor int32) (map[string]int32, error) {
```

Comment:
Instead of getting the new scale factor map.. get the map of existing scale factors.. change them in the torpedo test and then send it to scaleapplications()
That way the factor could be anything the test wants.. it does not have to be an addition.. it could be 2x.. let the test decide what arithmetic it wants to do

Reply:
If we are letting the test do the arithmetic, then we need not have a separate scale factor map. I thought the reason to have the map was to move the arithmetic outside the test.